astropy/visualization/wcsaxes/core.py

```diff
@@ -304,14 +304,17 @@ def plot_coord(self, *args, **kwargs):
             # Transform to the native frame of the plot
             frame0 = frame0.transform_to(native_frame)
 
+            coord_types = [x.coord_type for x in self.coords]
             plot_data = []
-            for coord in self.coords:
-                if coord.coord_type == 'longitude':
-                    plot_data.append(frame0.spherical.lon.to_value(u.deg))
-                elif coord.coord_type == 'latitude':
-                    plot_data.append(frame0.spherical.lat.to_value(u.deg))
-                else:
-                    raise NotImplementedError("Coordinates cannot be plotted with this "
+            if 'longitude' and 'latitude' in coord_types:
+                for coord in self.coords:
```

Comment:
This is now making me think that maybe the ordering of the data in `plot_data` matters, it would probably need to match the ordering of the `coord_types`. I sense we need a whole bunch of better test cases for this. :thinking:

Reply:
Hi @Cadair, the ordering in `coord_types` looks unimportant as it seems to be used only for checking whether longitude and latitude exist in the list. Might as well use a set instead for `coord_types`.

In case you mean that the ordering of `plot_data` should match the one in `self.coords`, I believe the code ensures that just as it did so before the commit.